Framework/PythonInterface/plugins/algorithms/BilbySANSDataProcessor.py

```diff
@@ -0,0 +1,603 @@
+            if time_mode:
+                ws_blk = self._convert_units(ws_blk, "Wavelength")
+            ws_blk = self._rebin(ws_blk, binning_wavelength, preserveevents=False)
+            # estimated blk counts for given measurement time and bin width
+            ws_blk_est = self._multiply(ws_blk, ws_blk_scaling)
+            ws_sam = self._subtract(ws_sam, ws_blk_est)
+
+        # sensitivity
+        pixeladj = ws_sen
+
+        ws_tran = self._emp_shape_adjustment(ws_tran,
+                                             ws_emp)  # swap arrays; ws_emp is always be shorter or equal to ws_tran
+        wavelengthadj = self._multiply(ws_emp, ws_tran)
+
+        # calculate the wide angle correction for sample transmission
+        if wide_angle_correction:
+            wavepixeladj = self._wide_angle_correction(ws_sam, ws_tran)
+        else:
+            wavepixeladj = None
+
+        # distance to maximum of parabolic motion of neutrons
+        real_l1 = self._get_l1(ws_sam)  # distance from the end of the last guide to the sample
+        extralength = 0.5 * real_l1  # neutrons following parabolic trajectory with maximum at the middle of the L1
+
+        # normalize vector to counting time of sample & long empty beam run
+        ws_emp_time = self._get_frame_count(ws_emp)
+        ws_sam_time = self._get_frame_count(ws_sam)
+        scale_full = scale * (ws_emp_time / ws_sam_time)
+        # extra multiplier is needed because measured transmission is ~5% lower;
+        # we need to divide result by lower number, hence need to lowering the final result, i.e. divide by 1.05
+        f = self._single_valued_ws(scale_full / (thickness * 1.05))
+
+        if reduce_2d:
+            q_max = binning_q[2]
+            q_delta = binning_q[1]
+            qxy = self._qxy(ws_sam, q_max, q_delta, pixeladj, wavelengthadj, account_for_gravity, solid_angle_weighting,
+                            extralength)
+            qxy = self._multiply(qxy, f)
+            self.setProperty("OutputWorkspace", qxy)
+        else:
+            deltar = 5.0  # Virtual ring width on the detector (mm). Hardcoded. Not sure it is a good way to go.
+
+            if (ws_sam.run().getProperty("source_aperture").value):
+                sourceapertureradius = float(ws_sam.run().getProperty("source_aperture").value) / 2.0
+                if sourceapertureradius > 40.0:
+                    sourceapertureradius = 20.0
+                    print("sourceapertureradius value cannot be retrieved; generic value of 20mm taken")
+            else:
+                sourceapertureradius = 20.0  # radius in mm
+                print("sourceapertureradius value cannot be retrieved; generic value of 20mm taken")
+
+            if (ws_sam.run().getProperty("sample_aperture").value):
+                sampleapertureradius = float(ws_sam.run().getProperty("source_aperture").value) / 2.0
+                if sampleapertureradius > 40.0:
+                    sampleapertureradius = 6.25
```

Comment:
Again it might be better to set these default values somewhere more visible, or even in a parameters file?